File: Frontend/src/model/product.ts
export interface ProductRequest {
  name: string;
  brand: string;
  description: string;
  specification: string;
  categoryId: number;
  netPrice: number;
  sellPrice: number;
  quantityInStock: number;
}

export type ProductResponse = ProductRequest;

export interface Product extends ProductRequest {
  id: number;
  createdDate: Date;
  updatedDate: Date;
}

File: Backend/Models/Product.cs
using System.Text.Json.Serialization;
using PhoneVault.Models;

namespace PhoneVault.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Brand { get; set; }
        public string Description { get; set; }
        public string Specification { get; set; }
        public decimal NetPrice { get; set; }
        public decimal SellPrice { get; set; }
        public int QuantityInStock { get; set; }
        public DateTime? CreatedDate { get; set; } = DateTime.Now;
        public DateTime? UpdatedDate { get; set; } = DateTime.Now;
        public int CategoryId { get; set; }

        [JsonIgnore] public Category? Category { get; set; } = null;
        [JsonIgnore] public ICollection<Review> Reviews { get; set; } = [];
    }
}